TTITransition/ViewControllers/NewsTableViewController.swift

```diff
@@ -63,30 +89,41 @@ class NewsTableViewController: UIViewController, UITableViewDataSource, UITableV
         }
     }
     
-    
     func createdIcons(fromNews: [NewsItem]) {
-        
+        for i in allNews {
+            let itm = NewsIcon(from: i.url, andDelegegate: self)
+            itm.delegateUpdateIcon = self
+            allIcons.append(itm)
+        }
+        mainQueue.async {
+            if !self.allNews.isEmpty {
+                self.activityIndicator.stopAnimating()
+            }
+        }
     }
+    
     @IBAction func segmentControl(_ sender: UISegmentedControl) {
         newsService.cancelCurrentDownloading()
-        self.switchLoadIndicator()
-        self.activityIndicator.startAnimating()
         allIcons = []
+        numberNewsUploaded = 20
         switch newsTypeSelector.selectedSegmentIndex {
         case 0:
             utilityQueue.async { [weak self] in
                 guard let strongSelf = self else { return }
-                strongSelf.allNews = strongSelf.newsService.loadNewsItems(for: .new)
+                strongSelf.allNews = strongSelf.newsService.loadNewsItems(for: .new, howMuchMore
+: strongSelf.numberNewsUploaded)
             }
         case 1:
             utilityQueue.async { [weak self] in
                 guard let strongSelf = self else { return }
-                strongSelf.allNews = strongSelf.newsService.loadNewsItems(for: .top)
+                strongSelf.allNews = strongSelf.newsService.loadNewsItems(for: .top, howMuchMore
+: strongSelf.numberNewsUploaded)
             }
         case 2:
             utilityQueue.async { [weak self] in
                 guard let strongSelf = self else { return }
-                strongSelf.allNews = strongSelf.newsService.loadNewsItems(for: .best)
+                strongSelf.allNews = strongSelf.newsService.loadNewsItems(for: .best, howMuchMore
```

Comment:
var newsSource: NewsSource!

switch (segmentedControl) {
case 1: newsSource = .top
case 2: newsSource = .best
....
} 

utilityQueue.async { [weak self] in
                  guard let strongSelf = self else { return }	
	                  guard let strongSelf = self else { return }
  strongSelf.allNews = strongSelf.newsService.loadNewsItems(for: newsSource, howMuchMore
 strongSelf.numberNewsUploaded)              }		              }